lib/tenstorrent/bh_arc/cm2dm_msg.c

```diff
@@ -348,3 +354,47 @@ int32_t Dm2CmSendThermTripCountHandler(const uint8_t *data, uint8_t size)
 	UpdateTelemetryThermTripCount(therm_trip_count);
 	return 0;
 }
+
+int32_t Dm2CmWriteTelemetry(const uint8_t *data, uint8_t size)
+{
+	if (size != 33) {
+		return -1;
```

Comment:
Just a general rule of thumb, if our FW is able to return an int on error, we should use a negative error value if an error occurs or zero on success.

So e.g. `-EINVAL` or something along those lines.